lib/Core/Executor.cpp

```diff
@@ -3554,7 +3553,7 @@ void Executor::runFunctionAsMain(Function *f,
   if (ai!=ae) {
     arguments.push_back(ConstantExpr::alloc(argc, Expr::Int32));
     if (++ai!=ae) {
-      Instruction *first = static_cast<Instruction *>(f->begin()->begin());
+      Instruction *first = &*f->begin()->begin();
```

Comment:
For cases where the "thing" being dereferenced is not a simple identifier (e.g. `i`) could you but the expression in brackets. I.e.

```c++
Instruction *first = &*(f->begin()->begin());
```

My reasoning is I don't want to have to look up the operator precedence when reading the code. Adding brackets makes it clear what is going on.